### ISSUE TITLE
Presto returning incorrect results when filtering on character strings containing unicode character \u2019

### ISSUE THREAD

**Issue by ptkool** [prestodb/presto#12467]
When filtering on character strings containing the unicode character \u2019, Presto returns incorrect results, depending on the type of filter being applied. Simple equality comparisons always return an incorrect result, as well as simple comparisons involving the LIKE predicate.

To reproduce the problem, unzip the attachment containing a small parquet file and create an external table in Hive as follows:

```text
create external table bad_stats (c1 string) stored as parquet location 'file:///<path_to_unzipped_directory';
```
In Presto CLI, run the following queries:

```text
presto> select * from hive.default.bad_stats;
               c1
--------------------------------
 AAA
 I sell with a different system
 I’m selling, just not online
(3 rows)

presto> select * from hive.default.bad_stats where c1 = 'I’m selling, just not online';
 c1
----
(0 rows)

presto> select * from hive.default.bad_stats where c1 like 'I’m selling, just not online';
 c1
----
(0 rows)

presto> select * from hive.default.bad_stats where regexp_like(c1, 'I’m selling, just not online');
              c1
------------------------------
 I’m selling, just not online
(1 row)

presto> select * from hive.default.bad_stats where upper(c1) = 'I’M SELLING, JUST NOT ONLINE';
              c1
------------------------------
 I’m selling, just not online
(1 row)
```

[bad_stats.zip](https://github.com/prestodb/presto/files/2962890/bad_stats.zip)

**Comment by ptkool** [comment]
**Observation 1**

I dug into this a bit more, and determined that Presto returns incorrect results when predicate pushdown is involved. Blocks are being ignored due to the fact the statistics are incorrect. In this particular case, Parquet shows the maximum value of the block to be `I sell with a different system` - should be `'I’m selling, just not online'`.

**Observation 2**
Create another parquet file with just the following two values:

    `I sell with a different system`
    `I’m selling, just not online`

In Presto CLI, I'm seeing the following:

```
presto> select * from hive.default.bad_stats;

Query 20190313_171244_00015_jp2qd failed: Corrupted statistics for column "[c1] BINARY" in Parquet file "file:/Users/michaelstyles/data/bad_stats/part-00000-b1907f6e-377d-4d4c-a9a1-538b0962746e-c000.snappy.parquet": [min: I’m selling, just not online, max: I sell with a different system, num_nulls: 0]
```

**Comment by nezihyigitbasi** [comment]
/cc @zhenxiao 

@ptkool I didn't debug the issue, but the workaround is to set the session property `parquet_fail_with_corrupted_statistics` to `false`.

**Comment by zhenxiao** [comment]
nice catch, @ptkool give me a few days, I will take a look after getting my hands clean

**Comment by wenleix** [comment]
Assign to @zhenxiao  :)

**Comment by sczhaoqi** [comment]
@zhenxiao 

I encountered a similar case. The results in presto when filtering on character strings is less than same query in hive. But when using to_utf8 or other method transfer the filtering column, it's ok.

I think it must cause by the pushdown using bianary to compare, but appear to user is varchar. Some specical character or different encoding will be ignore. 

How about using varchar to compare, ranther than binary, to make sure the correct results

-Apologize for my poor English

**Comment by soojinj** [comment]
Hi wanted to follow up on this thread! Was there any update?

**Comment by skadilover** [comment]
Seems that this issue still  not fixed?

**Comment by tdcmeehan** [comment]
@skadilover you are correct.  Please let me know if you would like to work on this.  I've added this to our prioritized bug backlog as well.  Thanks!

**Comment by skadilover** [comment]
> @skadilover you are correct. Please let me know if you would like to work on this. I've added this to our prioritized bug backlog as well. Thanks!

I`d like to figure it out , however,  the problem here is that the binary-compare logic is not same with parquet stastic, I am not clear that how to fix it , maybe need some days to clear it .
